### DISCUSSION TITLE
Using Podman on Windows Server 2022

### DISCUSSION THREAD

**Discussion by userid0x0** [containers/podman#17194] [Q&A] [answered]
Hi,
I browsed the Internet for a while and did not find a clear answer. We are using Windows Server 2022 and want to run Linux Containers on this machine using `podman`. Is this supported?
We succeeded using WSL2 on this machine, I furthermore found this thread
  https://techcommunity.microsoft.com/t5/itops-talk-blog/using-wsl-2-on-windows-server-2022-to-run-linux-containers/ba-p/3624745
that it's possible to run docker within WSL2. If I am not wrong thats exactly how `podman` works (under the hood) on Windows 10. I tried the podman 3.4.1 installer which fails when trying to exchange the SSH keys while `podman machine init`.
If I see it correct it fails in
- https://github.com/containers/podman/blob/ce504bbfe3d9fed8073160bbb7f54a1a259ec662/pkg/machine/wsl/machine.go#L392
- https://github.com/containers/podman/blob/ce504bbfe3d9fed8073160bbb7f54a1a259ec662/pkg/machine/wsl/machine.go#L524
- https://github.com/containers/podman/blob/ce504bbfe3d9fed8073160bbb7f54a1a259ec662/pkg/machine/wsl/machine.go#L916
- https://github.com/containers/podman/blob/ce504bbfe3d9fed8073160bbb7f54a1a259ec662/pkg/machine/keys.go#L36
- https://github.com/containers/podman/blob/ce504bbfe3d9fed8073160bbb7f54a1a259ec662/pkg/machine/keys.go#L77

because the data exchange between WSL2 and Windows is based on the fact that the working directory opens up on WSL side (is mounted on `/mnt/c/Users/<...>/...`). Unfortunately the user folders are directory junctions (symlinks) to Volumes (no idea what this is) and are thus inaccessible from WSL.
```
C:\Users>dir c:\Users
 Volume in drive C has no label.
 Volume Serial Number is ????-????

 Directory of c:\Users

23/01/2023  17:29    <DIR>          .
??/??/????  ??:??    <DIR>          Administrator.???
...
23/01/2023  ??:??    <JUNCTION>     ame???? [\??\Volume{????????-????-????-????-????????????}\]
....
```

**Comment by userid0x0** [comment]
A little update. I succeeded to use `podman` within a Debian WSL2 machine on Windows Server 2022.
I had to use the `/root/bootstrap` trick https://github.com/containers/podman/blob/ce504bbfe3d9fed8073160bbb7f54a1a259ec662/pkg/machine/wsl/machine.go#L71 and started WSL via `wsl -u root -d Debian /root/bootstrap`

It would be really great to see that `podman` can be controlled from `podman.exe` as well.

**Comment by baude** [comment]
There is a nice podman.exe that performs the install and setup of things automatically.  Did you try the downloads from [here](https://github.com/containers/podman/releases/tag/v4.3.1)

**Comment by userid0x0** [comment]
Hi @baude,
as I tried to say the Podman Windows based initialization fails. I then tried to crosscheck if I am able to run podman within WSL2 (without the windows part). I will create a Issue lateron as I found the rootcause for the failing `podman.exe machine init` and a workaround.

**Comment by userid0x0** [comment, marked as answer]
I opened up a ticket https://github.com/containers/podman/issues/17209
